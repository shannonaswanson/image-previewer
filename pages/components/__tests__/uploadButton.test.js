import { render, screen, cleanup } from "@testing-library/react";
import renderer from 'react-test-renderer';
import UploadButton from "../uploadButton";

describe('UploadButton', () => {
    const accept = '*.png';
    afterEach(() => cleanup());

    test('Should render without crashing', () => {
        render(<UploadButton />);
    });

    test('Should match snapshot', () => {
        const jsonTree = renderer.create(<UploadButton accept={accept} />).toJSON();
        expect(jsonTree).toMatchSnapshot();
    });

    test('Should have accept attibute', () => {
        render(<UploadButton accept={accept} />);
        const uploadButtonElement = screen.getByTestId('upload-input');
        expect(uploadButtonElement).toHaveAttribute('accept', accept);
    })
});