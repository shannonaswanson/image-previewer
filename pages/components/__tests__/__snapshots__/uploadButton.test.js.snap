// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadButton Should match snapshot 1`] = `
<div>
  <button
    className="button__Button-sc-1apg6n4-0 iCoJXd"
    data-testid="upload-button"
    onClick={[Function]}
  >
    Upload
  </button>
  <input
    accept="*.png"
    data-testid="upload-input"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />
</div>
`;
